schema.go

```diff
@@ -1363,10 +1422,18 @@ func isValidDefault(schema Schema, def any) (any, bool) {
 			}
 		}
 		return def, found
-	case String, Bytes, Fixed:
+	case String:
 		if _, ok := def.(string); ok {
 			return def, true
 		}
+	case Bytes, Fixed:
+		// Spec: Default values for bytes and fixed fields are JSON strings,
+		// where Unicode code points 0-255 are mapped to unsigned 8-bit byte values 0-255.
+		if d, ok := def.(string); ok {
+			if b, ok := isValidDefaultBytes(d); ok {
+				return b, true
+			}
+		}
```

Comment:
If my understanding of [specification](https://avro.apache.org/docs/1.11.1/specification/#schema-record) is correct:

> Default values for bytes and fixed fields are JSON strings, where Unicode code points 0-255 are mapped to unsigned 8-bit byte values 0-255.

It would be nice to add some validation and return a `[]byte` instead of `string`.

Reply:
Could be. I would leave that for another PR though.

Reply:
`validateDefault`'s change seems required for the new default decoder implementation. I ended up keeping it in this PR.